src/react/GroupDevicesPage.js

```diff
@@ -44,13 +44,19 @@ export default class GroupDevicesPage extends React.Component {
     this.handleModalCancel = this.handleModalCancel.bind(this);
     this.removeDevice = this.removeDevice.bind(this);
 
+    this.state = {
+      isRemovingDevice: false,
+      isLoadingGroups: false
+    };
+
     this.loadGroups();
   }
 
   loadGroups() {
     this.state = {
       phase: "loading",
-      showModal: false
+      showModal: false,
+      isLoadingGroups: true
```

Comment:
this is already expressed by the state `phase: "loading"`
Also, we're loading devices here, not groups